modules/fluid_properties/unit/src/TemperaturePressureFunctionFluidPropertiesTest.C

```diff
@@ -98,16 +101,105 @@ TEST_F(TemperaturePressureFunctionFluidPropertiesTest, properties)
   ABS_TEST(_fp->mu_from_v_e(v, e), visc, tol);
 }
 
+/**
+ * Verify calculation of the fluid properties with a function cp
+ */
+TEST_F(TemperaturePressureFunctionFluidPropertiesTest, properties_function_cp)
+{
+  const Real tol = REL_TOL_CONSISTENCY;
+  // Simpson integration only does 1e-5
+  const Real simpson_tol = 2e5 * tol;
```

Comment:
this is rough but yeah 3/8 simpson is there in accuracy. Maybe we should have a tunable method with a n-point quadrature instead.

Reply:
Also here when we deviate from one of our typical tolerances, I think it's preferred to not express it as a product but just write directly

Reply:
ok I reworked that. Simpson was doing a good job but with a parameter for the integration we can now do better than this. p_from_v_e is still challenging tolerance-wise but I dont think it's incorrect